azurelinuxagent/common/osutil/default.py

```diff
@@ -63,6 +67,40 @@ def __init__(self):
         self.selinux = None
         self.disable_route_warning = False
 
+    def enable_firewall(self, dst_ip=None, uid=None):
+        try:
+            if dst_ip is None or uid is None:
+                raise Exception("Missing arguments to enable_firewall")
+
+            # If either firewall rule exists, make no changes
+            accept_rule = FIREWALL_ACCEPT.format("C", dst_ip, uid)
+            drop_rule = FIREWALL_DROP.format("C", dst_ip)
+
+            if shellutil.run(accept_rule, chk_err=False) == 0 or \
+                shellutil.run(drop_rule, chk_err=False) == 0:
+                return
+
+            # Neither rule exists, append both rules
+            accept_rule = FIREWALL_ACCEPT.format("A", dst_ip, uid)
+            drop_rule = FIREWALL_DROP.format("A", dst_ip)
+
+            if shellutil.run(accept_rule) != 0:
+                logger.warn("Unable to add ACCEPT firewall rule '{0}'".format(
+                    accept_rule))
+                return
+
+            if shellutil.run(drop_rule) != 0:
+                logger.warn("Unable to add DROP firewall rule '{0}'".format(
+                    drop_rule))
+                return
+
+            rc, output = shellutil.run_get_output(FIREWALL_LIST)
+            logger.info("Successfully added firewall rules:\n{0}".format(
```

Comment:
Do we need to check the return code? I assume there are reasons the list command might fail